### PR TITLE
Updated meta-tx work

### PR DESCRIPTION
**Pull request by naterush**
This pull request simplifies the meta-tx pull request dramatically (and contains both the IdentityManager from the MetaIdentityManager). For the write-up, still see here: https://consensys.quip.com/CoClAjSTyBZI

### CODE REVIEW

**Comment by naterush** [comment]
@pelle quick note: I'll get 1. tests working, and 2. update MetaIdentityManager to take into account the recent changes in IdentityManager. Should be pretty simple. I'll update you when ready for reveiw :). Thanks!

File: test/txRelay.js [hunk 1]
```diff
@@ -0,0 +1,872 @@
...
+  } else {
+    data = '0x' + leftPad(n, '64', '0')
+    assert.equal(data.length, 66, "packed incorrectly")
+    return data;
+  }
+}
+
+function signPayload(signingAddr, sendingAddr, txRelay, destinationAddress, functionName,
+                     functionTypes, functionParams, lw, keyFromPw) {
+    return new Promise(
```
Comment (oed) [inline test/txRelay.js]: await could be used in this function instead.

File: test/txRelay.js [hunk 2]
```diff
@@ -0,0 +1,872 @@
...
+        recoveryKey = '0x'+acct[8]
+        recoveryKey2 = '0x'+acct[9]
+
+        sender = accounts[0]
+        notSender = accounts[1]
+        regularUser = accounts[2]
+
+        errorThrown = false
+
+        MetaTxRelay.new().then(instance => {
```
Comment (oed) [inline test/txRelay.js]: using await should work here as well

**Comment by oed** [comment]
Also: we need to add `MetaIdentityManager` and `TxRelay` to `index.js`

**Comment by naterush** [comment]
@oed changed tests to await. Added files to index.js

File: contracts/MetaIdentityManager.sol [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  mapping(address => uint) migrationInitiated;
+  mapping(address => address) migrationNewAddress;
+
+  modifier onlyAuthorized() {
+    if (msg.sender == relay || checkMessageData(msg.sender)) _;
+    else throw;
+  }
+
+  modifier onlyOwner(address identity, address sender) {
+    if (owners[identity][sender] > 0 && (owners[identity][sender] + userTimeLock) <= now ) _ ;
```
Comment (oed) [inline contracts/MetaIdentityManager.sol]: Do we want to use `isOwner` here?

Comment (naterush) [inline contracts/MetaIdentityManager.sol]: Yes, I think this makes sense for clarity. I will update now. Here and in IdentityManager

**Comment by christianlundkvist** [comment]
Ok, merging! 😃

**Pull request merged**